src/mumble/JackAudio.cpp

```diff
@@ -116,68 +125,57 @@ JackAudioSystem::JackAudioSystem()
 	} else {
 		iOutPorts = g.s.qsJackAudioOutput.toInt();
 	}
-	memset((void*)&out_ports, 0, sizeof(out_ports));
+	memset(reinterpret_cast<void *>(&out_ports), 0, sizeof(out_ports));
 
 	qhInput.insert(QString(), tr("Hardware Ports"));
 	qhOutput.insert(QString::number(1), tr("Mono"));
 	qhOutput.insert(QString::number(2), tr("Stereo"));
-}
-
-void JackAudioSystem::init_jack() {
-	if (bJackIsGood) {
-		return;
-	}
 
-	output_buffer = NULL;
 	jack_status_t status = static_cast<jack_status_t>(0);
 	int err = 0;
 
 	jack_options_t jack_option = g.s.bJackStartServer ? JackNullOption : JackNoStartServer;
 	client = jack_client_open(g.s.qsJackClientName.toStdString().c_str(), jack_option, &status);
 
-	if (client) {
-		qWarning("JackAudioSystem: client \"%s\" opened successfully", jack_get_client_name(client));
-		iBufferSize = jack_get_buffer_size(client);
-		iSampleRate = jack_get_sample_rate(client);
-
-		err = jack_set_process_callback(client, process_callback, this);
-		if (err != 0) {
-			qWarning("JackAudioSystem: unable to set process callback - jack_set_process_callback() returned %i", err);
-			close_jack();
-			return;
+	if (!client) {
+		QStringList errors = jackStatusToStringList(status);
+		qWarning("JackAudioSystem: unable to open client due to %i errors:", errors.count());
+		for (int i = 0; i < errors.count(); ++i) {
+			qWarning("JackAudioSystem: %s", qPrintable(errors.at(i)));
 		}
+	}
 
-		err = jack_set_sample_rate_callback(client, srate_callback, this);
-		if (err != 0) {
-			qWarning("JackAudioSystem: unable to set sample rate callback - jack_set_sample_rate_callback() returned %i", err);
-			close_jack();
-			return;
-		}
+	qWarning("JackAudioSystem: client \"%s\" opened successfully", jack_get_client_name(client));
```

Comment:
I get a segfault after these changes when jackd fails to launch :

```#0  0x00007f483abe8e3a in __strlen_sse2 () from /usr/lib/libc.so.6
<W>2019-05-03 23:44:35.830 JackAudioSystem: unable to open client due to 2 errors:
<W>2019-05-03 23:44:35.830 JackAudioSystem: JackFailure - overall operation failed
<W>2019-05-03 23:44:35.830 JackAudioSystem: JackServerFailed - unable to connect to the JACK server

Thread 1 "mumble" received signal SIGSEGV, Segmentation fault.
```

I would think that this is the message responsible, but I'm rusty with gdb and I can't find right now how to get line numbers in debugging symbols.

Reply:
You can pass the `symbols` CONFIG option to qmake.

Reply:
Thanks, that's what I tried but I put a `+` in front...
So the stacktrace confirmed my assumption. Putting a `return` at the end of the `if (!client)` [block above](https://github.com/mumble-voip/mumble/pull/3683/files#diff-34d235dc674ee341f5419adaf619b897R140) solves this segfault.

```
#0  0x00007ffff5876e3a in __strlen_sse2 () from /usr/lib/libc.so.6
#1  0x00007ffff5e8b448 in QString::vasprintf(char const*, __va_list_tag*) () from /usr/lib/libQt5Core.so.5
#2  0x00007ffff5dfe5ec in ?? () from /usr/lib/libQt5Core.so.5
#3  0x00007ffff5dcd087 in QMessageLogger::warning(char const*, ...) const () from /usr/lib/libQt5Core.so.5
#4  0x00005555557e0a78 in JackAudioSystem::JackAudioSystem (this=0x7fffe4014390) at JackAudio.cpp:148
#5  0x00005555557e2dfe in JackAudioInit::initialize (this=0x555555d8a2d0 <jackinit>) at JackAudio.cpp:83
#6  0x00005555556f3006 in DeferInit::run_initializers () at Global.cpp:196
#7  0x000055555565b701 in main (argc=1, argv=0x7fffffffe658) at main.cpp:366
```

Reply:
I accidentally forgot to put it, sorry.

Thank you very much for testing the pull request!